### PR TITLE
Scalar preferences

### LINKED ISSUE #1442: Problem colouring mesh by cells when n. of faces == n. of vertices
**Describe the bug, what's wrong, and what you expect:**

Some meshes such as a tetrahedron have same number of vertices and faces. Vertex shading is always preferred over cells in such a case but user should be able to choose which method to be preferred.

In the meantime I'm using some dirty workaround like adding a redundant vertex to vertices, but would be glad if this behaviour can be user configured.

Maybe related:  https://github.com/pyvista/pyvista/blob/9def273bf7df5f51796173148f8d729bed64409e/pyvista/plotting/plotting.py#L1636 

-----

**To Reproduce**

```py
# Construct some mesh
# 4 verts, 4 faces
vertices = np.array([[0, 0, 0], [1, 0, 0], [1, 1, 0], [0, 0, 1]])
faces = np.hstack([[3, 0, 1, 2], [3, 0, 3, 2], [3, 0, 1, 3], [3, 1, 2, 3]])
mesh = pyvista.PolyData(vertices, faces)
colors = [
    [255, 0, 0],
    [0, 255, 0],
    [0, 0, 255],
    [255,255,255]
]

mesh.cell_arrays['colors'] = colors 
mesh.plot(scalars='colors', show_edges=True, rgb=True)
```

**Screenshots**
Above: what we get, below: what I want to achieve.

image

-----

**System Information:**

Running ubuntu 20.04 on Docker Desktop for Mac 3.3.3 (64133)

```txt
--------------------------------------------------------------------------------
  Date: Fri Jun 25 12:53:47 2021 UTC

                OS : Linux
            CPU(s) : 4
           Machine : x86_64
      Architecture : 64bit
       Environment : Jupyter
        GPU Vendor : Mesa/X.org
      GPU Renderer : llvmpipe (LLVM 11.0.0, 256 bits)
       GPU Version : 4.5 (Core Profile) Mesa 20.2.6

  Python 3.8.5 (default, May 27 2021, 13:30:53)  [GCC 9.3.0]

           pyvista : 0.31.3
               vtk : 9.0.1
             numpy : 1.21.0
           imageio : 2.9.0
           appdirs : 1.4.4
            scooby : 0.5.7
            meshio : 4.4.6
        matplotlib : 3.4.2
           IPython : 7.24.1
             scipy : 1.7.0
              tqdm : 4.61.1
--------------------------------------------------------------------------------
```

### PR DESCRIPTION
**Pull request by akaszynski**
Resolves #1442

@kymok pointed out that in the edge case where `mesh.n_points == mesh.n_cells`, it's impossible to get cells to be prioritized when plotting rgb scalars (really, any scalars).

This PR adds an additional kwarg to `add_mesh` to allow the prioritization of cells rather than the default points.

### CODE REVIEW

File: pyvista/plotting/plotting.py [hunk 1]
```diff
@@ -1279,9 +1279,15 @@ def add_mesh(self, mesh, color=None, style=None, scalars=None,
             nonnegative, if supplied. If ``None``, the magnitude of
             the vector is plotted.
 
+        scalar_priority_cells : bool, optional
+            When ``mesh.n_points == mesh.n_cells`` and setting
+            scalars, this parameter sets how the scalars will be
+            mapped to the mesh.  Default ``False``, causing the
+            scalars will be associated with the mesh points points.
```
Comment (tkoyama010) [inline pyvista/plotting/plotting.py:1286]:
```suggestion
            scalars will be associated with the mesh points.
```

**Review by tkoyama010 — APPROVED** [review]
LGTM with 1 small suggestion.

**Comment by MatthewFlamm** [comment]
Can this be collapsed with the `preference` argument?  It is currently undocumented.

**Comment by akaszynski** [comment]
That's better.  Want to implement that or should I?

**Comment by MatthewFlamm** [comment]
I won't have time to spearhead for a few weeks as I'm getting ready for vacation and I still have #1441 to work on.  I can probably review though.

**Comment by akaszynski** [comment]
> I won't have time to spearhead for a few weeks as I'm getting ready for vacation and I still have #1441 to work on. I can probably review though.

I've documented the `preference` kwarg and used it in place of `scalar_cell_priorities`.

**Pull request merged**